packaging/macos/sign_notarize_staple.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+DMG_FILE="${1}"
+[ -z "${DMG_FILE}" ] && echo "Pass DMG file name as first argument." >&2 && exit 1
+[ -z "${APPLE_CODESIGN_IDENTITY}" ] && echo "Please set the $APPLE_CODESIGN_IDENTITY env var." >&2 && exit 1
+[ -z "${APPLE_BUNDLE_ID}" ] && echo "Please set the $APPLE_BUNDLE_ID env var." >&2 && exit 1
+[ -z "${APPLE_ID_USERNAME}" ] && echo "Please set the $APPLE_ID_USERNAME env var." >&2 && exit 1
+[ -z "${APPLE_APP_SPECIFIC_PASSWORD}" ] && echo "Please set the $APPLE_APP_SPECIFIC_PASSWORD env var." >&2 && exit 1
+[ -z "${ASC_PROVIDER}" ] && echo "Please set the $ASC_PROVIDER env var." >&2 && exit 1
+
+echo "Signing $DMG_FILE"
+codesign --verbose=4 --options runtime \
+    --sign "${APPLE_CODESIGN_IDENTITY}" "$(dirname $0)/entitlements.plist" "${DMG_FILE}"
+
+echo "Notarizing $DMG_FILE"
+xcrun altool --notarize-app --primary-bundle-id "${APPLE_BUNDLE_ID}" --username "${APPLE_ID_USERNAME}" \
+    --password "${APPLE_APP_SPECIFIC_PASSWORD}" --asc-provider "${ASC_PROVIDER}" --file "${DMG_FILE}" \
+    --output-format xml > notarize_result.plist
+UUID="$(/usr/libexec/PlistBuddy -c 'Print notarization-upload:RequestUUID' notarize_result.plist)"
+echo "Notorization UUID: $UUID"
+rm notarize_result.plist
+
+# wait for confirmation that notarization finished
+while true; do
+    xcrun altool --notarization-info "$UUID" \
+    --username "${APPLE_ID_USERNAME}" --password "${APPLE_APP_SPECIFIC_PASSWORD}" \
+    --output-format xml > notarize_status.plist
+
+    if [ "$?" != "0" ]; then
```

Comment:
Here shellcheck complains:

```
In packaging/macos/sign_notarize_staple.sh line 29:
    if [ "$?" != "0" ]; then
         ^--^ SC2181: Check exit code directly with e.g. 'if mycmd;', not indirectly with $?.
```

I don't think that it's more readable though, so let's disable that check here:

```suggestion
    xcrun altool --notarization-info "$UUID" \
    --username "${APPLE_ID_USERNAME}" --password "${APPLE_APP_SPECIFIC_PASSWORD}" \
    --output-format xml > notarize_status.plist

    # shellcheck disable=SC2181
    if [ "$?" != "0" ]; then
```

Then this file would be shellcheck clean in case we merge #3492.